GPU/Vulkan/TextureCacheVulkan.cpp

```diff
@@ -1541,93 +1537,96 @@ void *TextureCacheVulkan::DecodeTextureLevel(GETextureFormat format, GEPaletteFo
 	case GE_TFMT_5551:
 	case GE_TFMT_5650:
 		if (!swizzled) {
-			int len = std::max(bufw, w) * h;
-			tmpTexBuf16.resize(len);
-			finalBuf = tmpTexBuf16.data();
-			ConvertColors(finalBuf, texptr, dstFmt, bufw * h);
+			// Just a simple copy, we swizzle the color format.
+			for (int y = 0; y < h; ++y) {
+				memcpy(out + outPitch * y, texptr + bufw * sizeof(u16) * y, w * sizeof(u16));
+			}
+			finalBuf = out;
 		} else {
 			tmpTexBuf32.resize(std::max(bufw, w) * h);
 			UnswizzleFromMem(tmpTexBuf32.data(), texptr, bufw, h, 2);
```

Comment:
I'm sure you've thought of this, but UnswizzleFromMem could use an outPitch parameter - with that, we could avoid the memcpy loop. I'm fine with leaving that for later though, of course.